### PR TITLE
Implement Get all mentor emails endpoint (Admin) #43

### LINKED ISSUE #43: Implement Get all mentor emails endpoint (Admin)
**Description:**
This issue involves implementing an API to Get all mentor emails. The endpoint should allow Admin to make a GET request to `{{baseUrl}}/api/admin/mentor/emails?status=ACCEPTED`.

```
{
    "emails": [
        "mentor@gmail.com",
        "anotherMentor@gmail.com"
    ]
}
```

**Tasks:**
1. Create a controller for the `/controllers/profile` endpoint in the backend (create the route email if not created).
2. Parse and validate the request body to ensure it matches the expected format.
3. Implement appropriate error handling and response status codes for different scenarios (e.g., validation errors, database errors).
5. Write unit tests to validate the functionality and correctness of the endpoint.

API documentation: https://documenter.getpostman.com/view/27421496/2s93m1a4ac#8744a3ee-970f-489a-853d-8b23fdee8de3

ER diagram: https://drive.google.com/file/d/11KMgdNu2mSAm0Ner8UsSPQpZJS8QNqYc/view 

**Acceptance Criteria:**
- The profile API endpoint `/controllers/profile` is implemented via a GET request.
- The request body is properly parsed and validated for the required format.
- Appropriate error handling is implemented, providing meaningful error messages and correct response status codes.
- Unit tests are written to validate the functionality and correctness of the endpoint.

**Additional Information:**
No

**Related Dependencies or References:**
No

### PR DESCRIPTION
**Pull request by sathudeva7**
## Purpose

The purpose of this PR is to fix #43

## Goals

This issue involves implementing an API to Get all mentor emails. The endpoint should allow Admin to make a GET request to {{baseUrl}}/api/admin/mentor/emails?status=ACCEPTED.

## Approach

```
     const allMentors: Mentor[] = await mentorRepository.find({
      where: status ? { state: status } : {},
      select: ['profile'],
      relations: ['profile', 'category']
    })

    const emails = allMentors.map((mentor) => mentor?.profile?.primary_email)
```

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/services/admin/mentor.service.ts [hunk 1]
```diff
@@ -75,3 +75,38 @@ export const getAllMentors = async (
...
+  emails?: string[]
+  message: string
+}> => {
+  try {
+    const mentorRepository = dataSource.getRepository(Mentor)
+
+    const allMentors: Mentor[] = await mentorRepository.find({
+      where: status ? { state: status } : {},
+      select: ['profile'],
+      relations: ['profile', 'category']
```
Comment (anjula-sack) [inline src/services/admin/mentor.service.ts:92]: We don't need the category relation here.

```suggestion
      relations: ['profile']
```

File: src/services/admin/mentor.service.ts [hunk 2]
```diff
@@ -75,3 +75,38 @@ export const getAllMentors = async (
...
+  statusCode: number
+  emails?: string[]
+  message: string
+}> => {
+  try {
+    const mentorRepository = dataSource.getRepository(Mentor)
+
+    const allMentors: Mentor[] = await mentorRepository.find({
+      where: status ? { state: status } : {},
+      select: ['profile'],
```
Comment (anjula-sack) [inline src/services/admin/mentor.service.ts:91]: Do we have to select the profile here? Doesn't it return the profile from the below relation? Can you check again?

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/services/admin/mentor.service.ts (hunk 2 above)
Comment (sathudeva7) [reply src/services/admin/mentor.service.ts]: fixed thanks

File: src/controllers/admin/mentor.controller.ts [hunk 3]
```diff
@@ -67,3 +68,33 @@ export const getAllMentorsByStatus = async (
...
+    }
+  } catch (err) {
+    if (err instanceof Error) {
+      console.error('Error executing query', err)
+      res
+        .status(500)
+        .json({ error: 'Internal server error', message: err.message })
+    }
+  }
+}
```
Comment (anjula-sack) [inline src/controllers/admin/mentor.controller.ts:100]: We have refactored the code with return types. Can you take a pull and refer to an existing controller and update this? @sathudeva7

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Review by anjula-sack — APPROVED** [review]
Thank you! @sathudeva7

**Pull request merged**